### PR TITLE
refactor(PX-4655) - use displayState for calculating statuses on Purchase app

### PR DESCRIPTION
**Pull request by oxaudo**
This should have no changes on user front but does shuffles code around. Tagging couple of initial developers here for a review.

### CODE REVIEW

File: src/v2/Apps/Purchase/Components/OrderRow.tsx [hunk 1]
```diff
@@ -16,14 +15,14 @@ import { DateTime } from "luxon"
 import * as React from "react"
 import { createFragmentContainer, graphql } from "react-relay"
 import styled from "styled-components"
+import { RouterLink } from "v2/System/Router/RouterLink"
```
Comment (oxaudo) [inline src/v2/Apps/Purchase/Components/OrderRow.tsx:18]: This RouterLink is what we now have to use if I got comments correctly:)

File: src/v2/Apps/Purchase/Components/OrderRow.tsx [hunk 2]
```diff
@@ -38,15 +37,21 @@ const StyledImage = styled(Image)`
   width: 50px;
 `
 
+const UnderlineOnHoverLink = styled(RouterLink)`
```
Comment (oxaudo) [inline src/v2/Apps/Purchase/Components/OrderRow.tsx:40]: As far as I understood this functionality is not provided out of the box. A bit unclear if this page needs this one off look for links or we need to add this as an option to RouterLink / Link? For now just doing it inline here.

File: src/v2/Apps/Purchase/Components/OrderRow.tsx [hunk 3]
```diff
@@ -38,15 +37,21 @@ const StyledImage = styled(Image)`
...
+  text-decoration: none;
+  &:hover {
+    text-decoration: underline;
+  }
+`
+
 const OrderRow: React.FC<OrderRowProps> = props => {
   const { order } = props
+  const displayState = order.displayState
```
Comment (oxaudo) [inline src/v2/Apps/Purchase/Components/OrderRow.tsx:49]: This is the main refactor here. We've introduced `displayState` to order in Exchange that makes all those views much more streightforward.

**Review by oxaudo — COMMENTED** [review]
Some inline comments here

File: src/v2/Apps/Purchase/Components/OrderRow.tsx (hunk 2 above)
Comment (annacarey) [reply src/v2/Apps/Purchase/Components/OrderRow.tsx:40]: Maybe mention this briefly in #design-system?

File: src/v2/Apps/Purchase/__tests__/orderHelper.jest.tsx [hunk 4]
```diff
@@ -1,93 +1,48 @@
...
+describe("getOrderDisplayStateString", () => {
+  it.each([
+    ["SUBMITTED", "Pending"],
+    ["APPROVED", "Confirmed"],
+    ["PROCESSING", "Processing"],
+    ["IN_TRANSIT", "In transit"],
+    ["CANCELED", "Canceled"],
+    ["FULFILLED", "Delivered"],
+    ["REFUNDED", "Refunded"],
+  ])(
```
Comment (annacarey) [inline src/v2/Apps/Purchase/__tests__/orderHelper.jest.tsx:23]: NICE structure!

**Review by annacarey — APPROVED** [review]
Looks fine to me but I do think you should ask about that `RouterLink` styling thing in #design-system.

**Review by damassi — APPROVED** [review]

**Review by Serge0n — APPROVED** [review]

**Comment by oxaudo** [comment]
@annacarey - I think I'm going to go ahead and merge it as @damassi approved it here. Will mention the link issue in design-system as you've suggested though. We can follow up with another refactor later if desired.

**Pull request merged**